src/AwkwardArray.jl

```diff
@@ -293,7 +295,111 @@ function end_list!(layout::ListOffsetArray)
     layout
 end
 
-### ListOffsetArray with behavior = :string ##############################
+### ListArray ############################################################
+
+struct ListArray{INDEX<:IndexBig,CONTENT<:Content,BEHAVIOR} <: ListType{BEHAVIOR}
+    starts::INDEX
+    stops::INDEX
+    content::CONTENT
+    parameters::Parameters
+    ListArray(
+        starts::INDEX,
+        stops::INDEX,
+        content::CONTENT;
+        parameters::Parameters = Parameters(),
+        behavior::Symbol = :default,
+    ) where {INDEX<:IndexBig,CONTENT<:Content} =
+        new{INDEX,CONTENT,behavior}(starts, stops, content, parameters)
+end
+
+ListArray{INDEX,CONTENT}(;
+    parameters::Parameters = Parameters(),
+    behavior::Symbol = :default,
+) where {INDEX<:IndexBig} where {CONTENT<:Content} =
+    ListArray(INDEX([]), INDEX([]), CONTENT(), parameters = parameters, behavior = behavior)
+
+function copy(
```

Comment:
is this suppose to be `Base.copy`? I guess there's also `Base.deepcopy()` so it's one of them probably

Reply:
It's supposed to be distinct from `Base.copy`. Layout nodes in Awkward each have a constructor named `copy` that shallow-copies every argument that is not named: it's a great way to get "a struct exactly like `this_one` but with `this_field` and `that_field` replaced."

Before the final version of this package, the name might change to something that doesn't sound like it's supposed to be `Base.copy`. Or if there's a Julia way of doing this, it would be replaced by that. (Note that this `copy` can change type parameters as well as values.)